pkg/operator/client/client.go

```diff
@@ -64,6 +67,11 @@ func init() {
 	if err := maistrav2.SchemeBuilder.AddToScheme(scheme); err != nil {
 		panic(err)
 	}
+
+	// This is required because controller-runtime expects its consumers to
+	// set a logger through log.SetLogger within 30 seconds of the program's
+	// initalization.
+	ctrlruntimelog.SetLogger(logr.New(ctrlruntimelog.NullLogSink{}))
```

Comment:
`pkg/operator/client` package is used in the operator and e2e test binaries. The e2e binary was missing the logger so this change makes the required migration for it. However the operator binary already sets the controller runtime logger [here](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/operator.go#L62) so this change doesn't cause any conflict (because client's init runs first) but is redundant.
The alternative would be to add `SetLogger` somewhere in `test/e2e`.

Reply:
I don't see where `pkg/operator/client` is used by the operator, hence it seemed safe putting here. Am I misunderstanding the usage of `pkg/operator/client`?

Reply:
`pkg/operator/client` is imported in [pkg/operator/operator.go](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/operator.go#L24), that is, its `init` function was invoked.

Reply:
Oh whoops, you are right. I didn't give Goland enough time to finish loading all of the references...

Yea, I see, it's not-problematic, but redundant. I'm going to find a different place for it in `test/e2e`.